File: works/2579.java
import java.util.*;

public class Main
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int[] score = new int[302];
        int[][] d = new int[302][2];

        for(int i = 1; i <= n; i++)
            score[i] = sc.nextInt();

        score[0] = 0;
        d[0][0] = 0;
        d[0][1] = 0;
        d[1][0] = score[1];
        d[1][1] = score[1];

        for(int i = 2; i <= n; i++)
        {
            d[i][0] = d[i-1][1] + score[i];
            d[i][1] = Math.max(d[i-2][0], d[i-2][1]) + score[i];
        }

        System.out.println(Math.max(d[n][0], d[n][1]));
    }
}

File: works/2579.kt
import java.util.*;


fun main(args: Array<String>)
{
    var d = Array<Array<Int>>(301) {Array(3){0} }
    var a = Array(301) {0}

    val sc = Scanner(System.`in`)
    val n = sc.nextInt();
    for(i in 0..n-1) a[i] = sc.nextInt()

    d[0][1] = a[0]
    d[1][1] = a[1]
    d[1][2] = a[0] + a[1]

    for(i in 2..n-1)
    {
        d[i][1] = Math.max(d[i-2][1], d[i-2][2]) + a[i]
        d[i][2] = d[i-1][1] + a[i]
    }
    print(Math.max(d[n-1][1], d[n-1][2]))
}